src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -60,22 +66,165 @@ public RecoveryPointResponse GetRecoveryPoint()
 
         public ProtectionPolicyResponse CreatePolicy()
         {
-            throw new NotImplementedException();
+            string policyName = (string)providerData.ProviderParameters[PolicyParams.PolicyName];
+            WorkloadType workloadType = (WorkloadType)providerData.ProviderParameters[PolicyParams.WorkloadType];
+            BackupManagementType backupManagementType = (BackupManagementType)providerData.ProviderParameters[
+                                                                              PolicyParams.BackupManagementType];
+            AzureRmRecoveryServicesRetentionPolicyBase retentionPolicy = (AzureRmRecoveryServicesRetentionPolicyBase)
+                                                 providerData.ProviderParameters[PolicyParams.RetentionPolicy];
+            AzureRmRecoveryServicesSchedulePolicyBase schedulePolicy = (AzureRmRecoveryServicesSchedulePolicyBase)
+                                                 providerData.ProviderParameters[PolicyParams.SchedulePolicy];
+            string resourceName = (string)providerData.ProviderParameters[PolicyParams.ResourceName];
+            string resourceGroupName = (string)providerData.ProviderParameters[PolicyParams.ResourceGroupName];
+
+
+            // do validations
+            ValidateAzureVMWorkloadType(workloadType);
+
+            // validate both RetentionPolicy and SchedulePolicy
+            ValidateAzureVMRetentionPolicy(retentionPolicy);
+            ValidateAzureVMSchedulePolicy(schedulePolicy);
```

Comment:
Add validation for weekly backup schedule, i.e., only weekly retentions are allowed for weekly policy.